src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -139,19 +140,36 @@ export class TerminalInstance implements ITerminalInstance {
 		this._initDimensions();
 		this._createProcess(this._shellLaunchConfig);
 		this._createXterm();
+		if (this._shellLaunchConfig) {
+			this.setTitle(path.basename(this._shellLaunchConfig.executable), true);
+		}
 
 		if (platform.isWindows) {
```

Comment:
But that's more for the comment above. This one is pointing at the `if (platform.Windows)` section being too large